sqsconsumer.go

```diff
@@ -192,6 +192,7 @@ func (m *SmartSQSConsumer) worker(ctx context.Context, messages <-chan *sqs.Mess
 				// check to see if we've reached the maximum number of retries allowed
 				receiveCount := getApproximateReceiveCount(message)
 				if receiveCount > m.MaxRetries {
+					m.GetSQSMessageConsumer().DeadLetter(ctx, message)
```

Comment:
This looks good, thanks for the changes. A non-rhetorical question for you though:

I'd expect that somewhere in either the implementation of `DeadLetter`, or after `m.GetSQSMessageConsumer().DeadLetter(ctx, message)`, we would ideally want to log an `id` for accessing whatever message was deadlettered. My question now is, should `DeadLetter` return an `id` for runsqs to log? Or should implementations of `DeadLetter` be responsible for their own logging of `id`. 

In such a sense, I'd prefer the first option and have runsqs log the `id` returned by `DeadLetter,` that way we never forget to log an id. Thoughts?

Reply:
Considering this is a library, I would want the library to be log-level configurable.  It's not configurable in its current form, and adding an info line here without any context (like logging each ConsumeMessage call with a count) would be hard to grok when reading the log output later.

So I guess what I'm saying is, a near-term future addition to the runsqs lib could/should be level-configurable logging, at which point we would put something like `log.Info(fmt.Sprintf("calling DeadLetter with message id %d after exhausting MaxRetries %d", id, maxRetries)`.

Suggest we do not add a log line in this PR, but put up a task to review/add logging throughout the lib in the near future.

Reply:
I'm cool with that. 👍